test/lib/meetup_test.rb

```diff
@@ -0,0 +1,118 @@
+    assert_equal 21767819, parsed_response.first['id']
+  end
+
+  test "group response code of 200 returns response" do
+    Meetup.stubs(:get).with("/pro/operationcode/groups", options).returns(group_endpoint_response)
+
+    assert_nothing_raised { Meetup.new.operationcode_data }
+  end
+
+  test "group response code not equal to 200 raises error" do
+    Meetup.stubs(:get).with("/pro/operationcode/groups", options).returns(build_response(400))
+
+    assert_raises(Exception) { Meetup.new.operationcode_data }
+  end
+
+  test "event response code of 200 returns response" do
+    url = "Operation-Code-Hampton-Roads"
+
+    Meetup.stubs(:get).with("/#{url}/events", options).returns(event_endpoint_response)
+
+    assert_nothing_raised { Meetup.new.get_events_for url }
+  end
+
+  test "event response code not equal to 200 raises error" do
+    url = "Operation-Code-Hampton-Roads"
+
+    Meetup.stubs(:get).with("/#{url}/events", options).returns(build_response(400))
+
+    assert_raises( Exception ) { Meetup.new.get_events_for url }
+  end
+
+  #There are 4 events in the stub data, 1 with no venue
+  test "Meetup events with no venue are not included" do
+    url = "Operation-Code-Hampton-Roads"
+
+    Meetup.stubs(:get).with("/pro/operationcode/groups", options).returns(group_endpoint_response)
+    Meetup.stubs(:get).with("/#{url}/events", options).returns(event_endpoint_response)
+
+    events = Meetup.new.event_details_by_group
+
+    assert_equal( events.length, 3 )
+  end
+
+  #Event endpoint stubbed reponse has 3 good event records.
+  test "good data saves as new database record" do
+    url = "Operation-Code-Hampton-Roads"
+
+    Meetup.stubs(:get).with("/pro/operationcode/groups", options).returns(group_endpoint_response)
+    Meetup.stubs(:get).with("/#{url}/events", options).returns(event_endpoint_response)
+
+    assert_difference 'Event.count', 3 do
+      Meetup.new.add_events_to_database!
+    end
+  end
+
+  test "duplicate data does not create new record" do
+    url = "Operation-Code-Hampton-Roads"
+
+    Meetup.stubs(:get).with("/pro/operationcode/groups", options).returns(group_endpoint_response)
+    Meetup.stubs(:get).with("/#{url}/events", options).returns(event_endpoint_response)
+
+    # create a Event record from first item in `event_endpoint_parsed_response`
+    assert_equal 0, Event.count
+    meetup = Meetup.new
+    attributes = meetup.send(:details_for, event_endpoint_parsed_response.first)
+    Event.create(attributes)
+    assert_equal 1, Event.count
+
+    Meetup.new.add_events_to_database!
+    assert_equal 3, Event.count
+  end
+
+  test "updated data is updated in database" do
+    url = "Operation-Code-Hampton-Roads"
+
+    Meetup.stubs(:get).with("/pro/operationcode/groups", options).returns(group_endpoint_response)
+    Meetup.stubs(:get).with("/#{url}/events", options).returns(event_endpoint_response)
+
+    # create a Event record from first item in `event_endpoint_parsed_response`
```

Comment:
For lines 89 - 102 or so, you can create a method to abstract this logic into.

Reply:
this still needs to be addressed

Reply:
I took the logic that was there and made the create_and_save_old_event method which I put at the end of the file.  Is that not what you meant?

Reply:
Nice 👍